p4c.bzl

```diff
@@ -12,41 +12,110 @@
-        ctx.file.p4program.basename[:-3],
-        ".json"
+        out_dir,
+        "/",
+        ctx.files.srcs[0].basename[:-3],
+        extension
     ])
 
-    json_output_file = ctx.actions.declare_file(
+    # declare the output file
+    output_file = ctx.actions.declare_file(
         fname,
-        sibling=ctx.file.p4program
-    )
-
-    out_dir = json_output_file.path[:-len(json_output_file.basename)-1]
+        sibling=ctx.files.srcs[0])
+    
     ctx.actions.run_shell(
-        inputs = [ctx.file.p4program],
-        outputs = [json_output_file],
+        inputs = ctx.files.srcs,
+        outputs = [output_file],
         use_default_shell_env = True,
-        command = "p4c --std p4_16 --target bmv2 --arch v1model -o $1 $2",
+        command = "p4c --std $1 --target $2 --arch $3 -o $4 $5 $6",
         arguments = [
-            out_dir,
-            ctx.file.p4program.path
+            ctx.attr.std,
+            ctx.attr.target,
+            ctx.attr.arch,
+            output_file.path[:-len(output_file.basename)-1],
+            ctx.attr.p4c_args,
+            " ".join([file.path for file in ctx.files.srcs])
         ]
     )
 
-    return [DefaultInfo(files = depset([json_output_file]))]
+    return [DefaultInfo(files = depset([output_file]))]
 
 run_p4c = rule(
-    doc = "Runs p4c to produce output compiled files according to given params",
+    doc = "Runs p4c to produce output files according to given params.",
     implementation = _run_p4c_impl,
     attrs = {
-        "p4program": attr.label(
-            doc = "Input .p4 program to compile into a bmv2 JSON",
+        "srcs": attr.label_list(
+            doc = "Input .p4 files to pass to p4c.",
             mandatory = True,
-            allow_single_file = True,
+            allow_empty = False,
+            allow_files = [".p4"]
+        ),
+        "target": attr.string(
+            doc = "The --target argument passed to p4c (default: bmv2).",
+            mandatory = False,
+            default = "bmv2"
+        ),
+        "arch": attr.string(
+            doc = "The --arch argument passed to p4v (default: v1model).",
+            mandatory = False,
+            default = "v1model"
+        ),
+        "std": attr.string(
+            doc = "The --std argument passed to p4v (default: p4_16).",
+            mandatory = False,
+            default = "p4_16"
+        ),
+        "p4c_args": attr.string(
+            doc = "Any additional command line arguments to pass to p4c.",
+            mandatory = False,
+            default = ""
+        ),
+        "output": attr.string(
+            doc = """
+                The path of the directory containing the output file, passed to
+                p4c via -o, relative to the path of the first input file.
+                If not provided, this defaults to an auto generated path, based
+                on the rule name.
+                The output file is not checked into the repo, it is only created
```

Comment:
What is and isn't checked in seems independent of the build rule that produces it. Lets remove that comment.